src/main/java/de/retest/web/selenium/RecheckDriver.java

```diff
@@ -0,0 +1,273 @@
+package de.retest.web.selenium;
+
+import java.util.List;
+import java.util.Set;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.HasCapabilities;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.html5.LocalStorage;
+import org.openqa.selenium.html5.Location;
+import org.openqa.selenium.html5.LocationContext;
+import org.openqa.selenium.html5.SessionStorage;
+import org.openqa.selenium.html5.WebStorage;
+import org.openqa.selenium.interactions.HasInputDevices;
+import org.openqa.selenium.interactions.HasTouchScreen;
+import org.openqa.selenium.interactions.Keyboard;
+import org.openqa.selenium.interactions.Mouse;
+import org.openqa.selenium.interactions.TouchScreen;
+import org.openqa.selenium.internal.FindsByClassName;
+import org.openqa.selenium.internal.FindsByCssSelector;
+import org.openqa.selenium.internal.FindsById;
+import org.openqa.selenium.internal.FindsByLinkText;
+import org.openqa.selenium.internal.FindsByName;
+import org.openqa.selenium.internal.FindsByTagName;
+import org.openqa.selenium.internal.FindsByXPath;
+
+import de.retest.ui.descriptors.Element;
+import de.retest.ui.descriptors.RootElement;
+
+public class RecheckDriver implements WebDriver, JavascriptExecutor, FindsById, FindsByClassName, FindsByLinkText,
+		FindsByName, FindsByCssSelector, FindsByTagName, FindsByXPath, HasInputDevices, HasCapabilities,
+		TakesScreenshot, LocationContext, WebStorage, HasTouchScreen {
+
+	private final WebDriver wrapped;
+	private RootElement lastExpectedState;
+	private RootElement lastActualState;
+
+	public RecheckDriver( final WebDriver wrapped ) {
+		this.wrapped = wrapped;
+	}
+
+	public void setLastExpectdState( final RootElement lastExpectedState ) {
+		this.lastExpectedState = lastExpectedState;
+	}
+
+	public void setLastActualState( final RootElement lastActualState ) {
+		this.lastActualState = lastActualState;
+	}
+
+	public WebElement findElement( final ByRetestId by ) {
+		if ( lastExpectedState == null ) {
+			throw new IllegalStateException(
+					"You must use the UnbreakableRecheckImpl and first check the state before being able to use the retestId locator." );
+		}
+		final Element searchedFor = by.findElement( lastExpectedState, lastActualState );
+		return wrapped.findElement( By.xpath( searchedFor.getIdentifyingAttributes().getPath() ) );
+	}
+
+	public WebElement findElementByRecheckId( final String recheckId ) {
+		return findElement( new ByRetestId( recheckId ) );
+	}
+
+	@Override
+	public WebElement findElement( final By by ) {
+		if ( by instanceof ByRetestId ) {
+			return findElement( (ByRetestId) by );
+		}
+		// TODO We can implement to find the element with the given value in the lastCheckedState
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=77&issues=AWaHMZUnlB6-zLfzofGC&open=AWaHMZUnlB6-zLfzofGC)

[ISSUE_KEY]: AWaHMZUnlB6-zLfzofGC